shell_substitution.h

```diff
@@ -0,0 +1,92 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define COMM_SIZE 6
+
+// Display result of the assignment
+struct evaluation_factors {
+    bool valid;
+    char *assignment;
+};
+
+bool is_print_command_valid(char *line)
+{
+    // NOTE: Need to handle single quote as invalid
+    char *command = strndup(line, COMM_SIZE);
+
+    if (!strcmp(command, "print ")) return true;
+    return false;
+}
+
+char *retrieve_string_inside_quotation_mark(char *line)
+{
+    // Retrieve the string inside the quotation marks
+    char *display_str = (char *)malloc (sizeof (char) * strlen(line));
+    for (int i = (COMM_SIZE + 1); i < (strlen(line)-1); i++) 
+        display_str[i-(COMM_SIZE + 1)] = line[i];
+    return display_str;
+}
+
+bool is_dollar_present(char *str)
+{
+    for (int i = 0; i < strlen(str); i++)
+        if (str[i] == '$') return true;
+    return false;
+}
+
+int get_index_of_char_in_string(char *assignment)
```

Comment:
What is the difference between get_index_of_char_in_string and get_index_of_character? I think you should create more general function and pass the character which you are interesting in.

Reply:
Ah yes. This part can be merged into one.